src/components/Switch/ColoredSwitch.test.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import ColoredSwitch from './ColoredSwitch.component';
+
+describe('<ColoredSwitch />', () => {
+  const mockHandler = jest.fn();
+  const props = {
+    checked: false,
+    toggleSwitch: mockHandler,
+    switchColor: 'white',
+    label: 'Dark Mode',
+  };
+  let wrapper;
+  beforeEach(() => {
+    wrapper = render(<ColoredSwitch {...props} />);
+  });
+  test('Render ColoredSwitch component', () => {
+    expect(wrapper).not.toBeNull();
+  });
+  test('renders by Dark Mode', () => {
+    const darkMode = screen.getByText(/Dark Mode/i);
+    expect(darkMode).toBeInTheDocument();
+  });
+  test('Render after click', () => {
+    const { getByRole } = wrapper;
+    const button = getByRole('checkbox');
+    fireEvent.click(button);
+    expect(mockHandler).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
very good test, good work!

Just a minor comment: there is not need to create a `wrapper` variable, you can just execute `render` on `beforeEach` and internally, react testing library will have the render result (that's why you get values when you run the queries like `getByText `).

You can do the next:

```js
beforeEach(() => {
     render(<ColoredSwitch {...props} />);
  });

 test('renders by Dark Mode', () => {
    // code...
  });

  test('Render after click', () => {
    // code...
  });
```

And that's it, you are already validating that the component is rendering